File: Greedy/Java/Minimum Number of Changes to Make Binary String Beautiful.java
// Problem link: https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful/description/
// Time: O(n)
// space: O(1)
class Solution {
    public int minChanges(String s) {
        int result = 0;

        for(int index = 0; index < s.length(); index += 2) {
            if(s.charAt(index) != s.charAt(index + 1)) {
                result++;
            }
        }

        return result;
    }
}

File: Greedy/Python/Minimum Number of Changes to Make Binary String Beautiful.py
# Problem link: https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful/description/
# Time: O(n)
# Space: O(1)
class Solution:
    def minChanges(self, s: str) -> int:
        result = 0

        for index in range(0, len(s), 2):
            if s[index] != s[index + 1]:
                result += 1
        
        return result
